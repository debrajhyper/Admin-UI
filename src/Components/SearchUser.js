import React from 'react'

const SearchUser = ({ searchResult, setSearchResult }) => {

    //  Method to set the search text entered by the user to the state using the setSearchResult() method.
    const searchHandler = event => {
        setSearchResult(event.target.value);
    };

    return (
        <input 
        data-testid="search-user"
        type="text" 
        placeholder="Search"
        className="float-right outline-none border-2 dark:border-gray-700 rounded pl-2 px-1 md:px-2 py-1 dark:bg-gray-800 hover:border-blue-light dark:hover:border-blue-dark focus:border-blue-light dark:focus:border-blue-dark" 
        value={searchResult}
        onChange={searchHandler} 
        />
    )
}

export default SearchUser
