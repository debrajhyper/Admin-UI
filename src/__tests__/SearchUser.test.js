import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import SearchUser from "../Components/SearchUser";

afterEach(cleanup);

test("render without crashing", () => {
    render(<SearchUser searchResult="abh" />);

    const SearchUserElement = screen.getByTestId("search-user");
    expect(SearchUserElement).toBeInTheDocument();
    expect(SearchUserElement).toContainHTML("<input");
})

test("snapshot", () => {
    const tree = renderer.create(<SearchUser searchResult="abh" />).toJSON();
    expect(tree).toMatchSnapshot();
})