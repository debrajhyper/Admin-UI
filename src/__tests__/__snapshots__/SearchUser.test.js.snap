// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<input
  className="float-right outline-none border-2 dark:border-gray-700 rounded pl-2 px-1 md:px-2 py-1 dark:bg-gray-800 hover:border-blue-light dark:hover:border-blue-dark focus:border-blue-light dark:focus:border-blue-dark"
  data-testid="search-user"
  onChange={[Function]}
  placeholder="Search"
  type="text"
  value="abh"
/>
`;
